### DISCUSSION TITLE
Dynamically identifying image generating nodes outside of runtime

### DISCUSSION THREAD

**Discussion by tachyon-beep** [Comfy-Org/ComfyUI#5262] [Q&A]
Hi all,

I'm writing [some code](https://github.com/tachyon-beep/comfyui-simplefeed) that shunts a copy of images generated by nodes down to an image feed at the bottom of the screen, it also includes a (somewhat hacky) filter system that uses names of sampler nodes and save nodes to let users select nodes to be filtered out (e.g., images from the start of the pipeline that haven't been detailed yet), load image nodes, etc. I'm trying to replace this manual system with one that provides a list of all the nodes that output images, offering dynamic support for custom nodes and removing the need to manually maintain the list of 'image producing nodes'.

You can only use the executed event to retrieve an image has output an image during execution. This is entirely reasonable behaviour, no problems so far:

```
onExecuted({ detail }) {
    if (!this.visible || !detail?.output?.images) {
      return;
    }
    //My image stuff goes here.
}
```
 
The issue is that when you query a node and ask it about its outputs, the result it gives you seems to have no correlation with whether it actually outputs an image at runtime.

For example: `return node.outputs.some((output) => output.type === 'IMAGE');` 

- returns **true** for a VAE decoder node, which **doesn't** output an image during the executed event.
- returns **false** for a saveImage node, which **does** output an image during the executed event.

This inconsistency makes life tough for beleagered image tray node developers (and probably others) who rely on nodes to provide accurate information about their outputs during runtime. I don’t know enough about the underlying architecture to say whether this is a bug or if there’s some deeper nuance I'm missing, but I’d really appreciate any insights or suggestions the community can offer.

Thanks for all your hard work on a remarkable piece of software!

**Comment by wibur0620** [comment]
I really appreciate your plugin and have come up with some improvement ideas during the usage process. Could you consider placing the image list in the position shown in the diagram? This design could further enhance the user's visual experience.

**Reply by tachyon-beep** [reply]
I'm always happy to take on feedback to make it more useful - which diagram are you referring to?

**Reply by wibur0620** [reply]
> I'm always happy to take on feedback to make it more useful - which diagram are you referring to?

![Uploading image.png…]()

**Comment by wibur0620** [comment]
![image](https://github.com/user-attachments/assets/edb1609b-8b89-442a-9b1b-2b4aed73c960)

**Reply by tachyon-beep** [reply]
Thanks, I'll take a look - can you add it as a request here as well: https://github.com/tachyon-beep/comfyui-simplefeed/issues
